vision/app.py

```diff
@@ -212,16 +241,21 @@ def run_video(self):
             cv2.createTrackbar('Upper V', 'Settings', self.settings['upper'][2], 255,
                                lambda val: self.update_thresh(False, 2, val))
 
+        # Initializes array that stores raw camera frame outputs in the BGR color space
         bgr = np.zeros(shape=(360, 640, 3), dtype=np.uint8)
+        # Initializes array that stores camera frame outputs in the HSV color space
         im = np.zeros(shape=(360, 640, 3), dtype=np.uint8)
 
         while True:
+            # Fetches the most recent frame from the camera stream
             bgr = camera.read()
 
             if bgr is not None:
+                # Resize raw camera frame and convert into HSV
                 im = cv2.cvtColor(cv2.resize(bgr, (640, 360), 0, 0), cv2.COLOR_BGR2HSV)
```

Comment:
Oh another thing. If I'm understanding correctly, here we're swapping the width and the height of the image. What's the reason for doing this?

Reply:
Hah yeah I noticed that too, and I'm not 100% sure why this is there. It looks like maybe one function takes (width, height) and the other takes (height, width)...